packages/block-editor/src/components/inserter/menu.js

```diff
@@ -61,9 +61,19 @@ function InserterMenu(
 	const [ filterValue, setFilterValue, delayedFilterValue ] =
 		useDebouncedInput( __experimentalFilterValue );
 	const [ hoveredItem, setHoveredItem ] = useState( null );
-	const [ selectedPatternCategory, setSelectedPatternCategory ] = useState(
-		__experimentalInitialCategory
+
+	const selectedPatternCategory = useSelect(
+		( select ) => {
+			const _selectedPatternCategory =
+				select( blockEditorStore ).selectedPatternCategory();
+			return _selectedPatternCategory
+				? _selectedPatternCategory
+				: __experimentalInitialCategory;
+		},
+		[ __experimentalInitialCategory ]
 	);
+	const { setSelectedPatternCategory } = useDispatch( blockEditorStore );
```

Comment:
To explain my other comment more: instead of dispatching `setSelectedPatternCategory` to the block store, maybe dispatch something to the interface package about the layout we want. In the interface package we don't care what category is selected. We care about the layout we want when we have a category selected.